R/cut.R

```diff
@@ -0,0 +1,105 @@
+#' Cut a numeric Crunch variable
+#' 
+#' crunch::cut() is equivalent to base::cut() except that it operates on 
+#' crunch variables instead of in-memory variables. The function divides the range of 
+#' x into intervals and codes the values in x according to which interval they fall. 
+#' The leftmost interval corresponds to level one, the next leftmost to level two and 
+#' so on.
+#' @param x A crunch variable of the class NumericVariable
+#' @param breaks Either a numeric vector of two or more unique cut points 
+#' or a single number (greater than or equal to 2) giving the number of intervals 
+#' into which x is to be cut.
+#' @param variable.name The name of the resulting case variable as a character string. 
+#' @param labels labels for the levels of the resulting category.
+#' By default, labels are constructed using interval notation.
+#' If labels = FALSE, simple integer codes are returned instead of a factor.
+#' @param include.lowest logical, indicating if an `x[i]` equal to the lowest 
+#' (or highest, for right = FALSE) `breaks` value should be included.
+#' @param right logical, indicating if the intervals should be closed on the right 
+#' (and open on the left) or vice versa.
+#' @param dig.lab	integer which is used when labels are not given. 
+#' It determines the number of digits used in formatting the break numbers.
+#' @param ordered_result	Ignored.
+#' @param ... further arguments passed to or from other methods.
+#'
+#'
+#' @return a Crunch VariableDefinition
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ds <- loadDataset("mtcars")
+#' ds$cat_var <- cut(ds$mpg, 3, variableName = "new_var")
+#' }
+#'  
+setMethod("cut", "NumericVariable", function(x, 
+                                             breaks, 
+                                             variable.name, 
+                                             labels = NULL,
+                                             include.lowest = FALSE,
+                                             right = TRUE,
+                                             dig.lab = 3,
+                                             ordered_result = FALSE, ...){
+  env <- environment()
+  if (length(breaks) == 1L) {
+    if (is.na(breaks) || breaks < 2L) {
+      halt("invalid number of intervals")
+    }
+    nb <- as.integer(breaks + 1) # one more than #{intervals}
+    dx <- diff(rx <- c(min(x, na.rm = TRUE), max(x, na.rm = TRUE)))
+    if (dx == 0) {
+      dx <- abs(rx[1L])
+      breaks <- seq.int(rx[1L] - dx/1000, rx[2L] + dx/1000,
+                        length.out = nb)
+    } else {
+      breaks <- seq.int(rx[1L], rx[2L], length.out = nb)
+      breaks[c(1L, nb)] <- c(rx[1L] - dx/1000, rx[2L] + dx/1000)
+    }
+  }else nb <- length(breaks <- sort.int(as.double(breaks)))
+  if (anyDuplicated(breaks)) halt("'breaks' are not unique")
+  if (is.null(labels)) {#- try to construct nice ones ..
```

Comment:
Consider factoring this labeling code out into another function. Will make it easier for us to test, and easier for us to decide to do something different. I'm not sure that set notation will resonate with our target users.